<?xml version="1.0"?>

<launch>
      <param name="use_sim_time" value="true"/>

<arg name="drone_name" default="iris"/>
  <group ns="$(arg drone_name)">

    <node name="training_action_interface_node" pkg="training_q_learning" type="training_action_interface.py" output="screen" >
        <param name = "publish_hz" value = "100" />
        <param name = "drone_name" value = "$(arg drone_name)" />
    </node> 


    <node name="controller_v_z" pkg="pid" type="controller" ns="pid_v_z" output="screen" >
        <param name="node_name" value="pid_v_z" />
        <param name="Kp" value="5.0" />
        <param name="Ki" value="10.0" />
        <param name="Kd" value="0" />
        <param name="upper_limit" value="10" />
        <param name="lower_limit" value="0" />
        <param name="windup_limit" value="10" />
        <param name="max_loop_frequency" value="1000.0" />
        <param name="min_loop_frequency" value="1.0" />
        <param name="topic_from_controller" value="/$(arg drone_name)/training_action_interface/control_effort/v_z/"/>
        <param name="topic_from_plant" value="/$(arg drone_name)/training_action_interface/state/v_z"/>
        <param name="setpoint_topic" value="/$(arg drone_name)/training_action_interface/setpoint/v_z"/>   
    </node>

    <node name="controller_yaw" pkg="pid" type="controller" ns="pid_yaw" output="screen" >
        <param name="node_name" value="pid_yaw" />

        <param name="Kp" value="8" />
        <param name="Ki" value="1.0" />
        <param name="Kd" value="0" />

        <param name="upper_limit" value="3.141592" />
        <param name="lower_limit" value="-3.141592" />
        <param name="windup_limit" value="5" />
        <param name="max_loop_frequency" value="1000.0" />
        <param name="min_loop_frequency" value="1.0" />
        <param name="topic_from_controller" value="/$(arg drone_name)/training_action_interface/control_effort/yaw"/>
        <param name="topic_from_plant" value="/$(arg drone_name)/training_action_interface/state/yaw"/>
        <param name="setpoint_topic" value="/$(arg drone_name)/training_action_interface/setpoint/yaw"/>   
    </node> 

  </group>  
</launch>
